sklearn/utils/estimator_checks.py

```diff
@@ -760,7 +763,13 @@ def _check_transformer(name, transformer_orig, X, y):
         # raises error on malformed input for transform
         if hasattr(X, 'T'):
             # If it's not an array, it does not have a 'T' property
-            assert_raises(ValueError, transformer.transform, X.T)
+            with assert_raises(ValueError, msg="The classifer {name} does not "
```

Comment:
*classifier

Reply:
But should be "transformer" right? "The transformer {} does not raise an error when the number of features in ``transform`` is different from the number of features in ``fit``"?